lib/auth/saml.go

```diff
@@ -326,7 +326,7 @@ func (a *AuthServer) validateSAMLResponse(samlResponse string) (*samlAuthRespons
 	}
 	assertionInfo, err := provider.RetrieveAssertionInfo(samlResponse)
 	if err != nil {
-		log.Warnf("Failed to retrieve SAML AssertionInfo from response: %v.", err)
+		log.Warnf("Received response with incorrect or no claims/attribute statements. Please check the identity provider configuration to make sure that mappings for claims/attribute statements are set up correctly. <See: https://gravitational.com/teleport/docs/ssh_okta/>. Failed to retrieve SAML AssertionInfo from response: %v.", err)
```

Comment:
Since this is a SAML response, I think we should link to a FAQ on https://gravitational.com/teleport/docs/enterprise/ssh_sso/ .   Can you also create a table on this page that shows our preferred preference for 3rd party providers. e.g. Okta = SAML,  GSuite = OIDC

Reply:
I don't think that we currently have anything set up for this but as a suggestion, if we're going to start putting static links to our docs in code then we should look into setting up some kind of redirection service and using links to that instead.

This will mean we can use a link like "https://go.gravitational.com/id/123" which will then automatically redirect to the correct docs page and doesn't rely on us updating the code if we ever change the path to the docs - we just have to maintain that link elsewhere. It also allows us to monitor how many people are clicking the link. Microsoft, Google and various other services do this and it seems smart.

I don't know how much effort it would be to set this up.

Reply:
I was just thinking the same thing for deep linking content in the docs. The main question is, do we buy or build?

Reply:
I think we are OK to deep link for now, but @aelkugia can you please make it the generic ssh_sso page vs ssh_okta . Then we'll merge this in.

Reply:
@benarent updated to the mentioned link. I can start exploring setting up some kind of redirection service as mentioned.